src/test/test-tpm2.c

```diff
@@ -69,4 +69,94 @@ TEST(tpm2_util_pbkdf2_hmac_sha256) {
+                                .parameters.eccDetail = {
+                                        .symmetric = {
+                                                .algorithm = TPM2_ALG_AES,
+                                                .keyBits.aes = 128,
+                                                .mode.aes = TPM2_ALG_CFB,
+                                        },
+                                        .scheme.scheme = TPM2_ALG_NULL,
+                                        .curveID = TPM2_ECC_NIST_P256,
+                                        .kdf.scheme = TPM2_ALG_NULL,
+                                },
+                        },
+                },
+                /* index 2 RSA SRK */
+                {
+                        .publicArea = {
+                                .type = TPM2_ALG_RSA,
+                                .nameAlg = TPM2_ALG_SHA256,
+                                .objectAttributes = TPMA_OBJECT_FIXEDTPM|TPMA_OBJECT_FIXEDPARENT|TPMA_OBJECT_SENSITIVEDATAORIGIN|TPMA_OBJECT_RESTRICTED|TPMA_OBJECT_DECRYPT|TPMA_OBJECT_USERWITHAUTH|TPMA_OBJECT_NODA,
+                                .parameters.rsaDetail = {
+                                        .symmetric = {
+                                                .algorithm = TPM2_ALG_AES,
+                                                .keyBits.aes = 128,
+                                                .mode.aes = TPM2_ALG_CFB,
+                                        },
+                                        .scheme.scheme = TPM2_ALG_NULL,
+                                        .keyBits = 2048,
+                                },
+                        },
+                },
+                /* Index 3 ECC SRK */
+                {
+                        .publicArea = {
+                                .type = TPM2_ALG_ECC,
+                                .nameAlg = TPM2_ALG_SHA256,
+                                .objectAttributes = TPMA_OBJECT_FIXEDTPM|TPMA_OBJECT_FIXEDPARENT|TPMA_OBJECT_SENSITIVEDATAORIGIN|TPMA_OBJECT_RESTRICTED|TPMA_OBJECT_DECRYPT|TPMA_OBJECT_USERWITHAUTH|TPMA_OBJECT_NODA,
+                                .parameters.eccDetail = {
+                                        .symmetric = {
+                                                .algorithm = TPM2_ALG_AES,
+                                                .keyBits.aes = 128,
+                                                .mode.aes = TPM2_ALG_CFB,
+                                        },
+                                        .scheme.scheme = TPM2_ALG_NULL,
+                                        .curveID = TPM2_ECC_NIST_P256,
+                                        .kdf.scheme = TPM2_ALG_NULL,
+                                },
+                        },
+                },
+        };
```

Comment:
hmm, these structures are the same on all archs, right?

so instead of defining the whole thing twice, maybe just put a list of sha hashes here and validate them? we have sha256 code in place anyway, so that would be easy?

i think that would be less ugly then to literally write down the whole structs twice

Reply:
I think this would be the first time I disagree, but if it's a show stopper I'll do it. So instead of visually being able to see the comparison to whats expected, and then understand the changes implications, it's to magic hashes that need to be explained.

Reply:
I added this commit on top of the base commit, we can make a call to take it as a separate change or squash it down, just let me know.

Reply:
hmm, so these structs, are not actually the same everywhere, are they? i.e. this is not what is sent over the wire 1:1, right? because if it was, then we'd have to do a lot of htole()/htobe() stuff here... But that means the hashes are not stable anyway, as they would be different on LE and BE machines.

so i guess scratch my hash-based checking idea anyway...

Reply:
> hmm, so these structs, are not actually the same everywhere, are they? i.e. this is not what is sent over the wire 1:1, right? because if it was, then we'd have to do a lot of htole()/htobe() stuff here... But that means the hashes are not stable anyway, as they would be different on LE and BE machines.
> 
> so i guess scratch my hash-based checking idea anyway...

Correct, scalars are host to big endian converted.